src/rpp/shm_mutex.cpp

```diff
@@ -0,0 +1,80 @@
+#if __linux__
+#include "shm_mutex.h"
+#include "debugging.h"
+#include <condition_variable>
+
+// TODO: throw instead of logging errors
```

Comment:
Well, throwing is also problematic. We have a lot of code that we don't want to crash because of somewhat optional mutexes. If the mutex really fails we can just log it and return.